homeassistant/components/plex/media_player.py

```diff
@@ -385,20 +242,34 @@ def name(self):
         return self._name
 
     @property
+    @needs_session
     def username(self):
         """Return the username of the client owner."""
-        return self._session_username
-
-    @property
-    def app_name(self):
-        """Return the library name of playing media."""
-        return self._app_name
+        return self.session.username
 
     @property
     def state(self):
         """Return the state of the device."""
         return self._state
 
+    @state.setter
+    def state(self, state):
+        """Set the state of the device."""
+        if state == "playing":
+            self._state = STATE_PLAYING
+        elif state == "paused":
+            self._state = STATE_PAUSED
+        elif state == "stopped":
+            self.session = None
+            self.force_idle()
```

Comment:
You are calling `force_idle` from a property setter, which then calls `dispatcher_send`. Please do not use a property setter if you have side effects.

Reply:
I was thinking `dispatcher_send` just sent an event to another listener instead of handling the scheduling itself. This can be moved out of the setter.

Reply:
We should drop using `setter`. Especially since `state` is an inherited property.